lib/miq_automation_engine/service_models/miq_ae_service_manageiq-providers-vmware-infra_manager-vm.rb

```diff
@@ -15,5 +15,9 @@ def add_disk(disk_name, disk_size_mb, options = {})
     def remove_disk(disk_name, options = {})
       sync_or_async_ems_operation(options[:sync], "remove_disk", [disk_name, options])
     end
+
+    def move_into_folder(folder, options = {})
+      sync_or_async_ems_operation(options[:sync], "move_into_folder", [folder.ems_ref])
```

Comment:
We should keep the check you had for EmsFolder type as many objects expose the `ems_ref` property and could lead to difficult to debug errors.  Would be nice to deal with an invalid object type before queuing the work.